eo-maven-plugin/src/main/java/org/eolang/maven/optimization/OptCached.java

```diff
@@ -113,16 +104,11 @@ private Path cached(final XML xml) {
      */
     private boolean contains(final XML xml) throws IOException {
         final Path path = this.cached(xml);
-        final Optional<String> time = xml.xpath("/program/@time").stream().findFirst();
+        final Optional<String> hash = xml.xpath("/program/@hash").stream().findFirst();
         final boolean res;
-        if (Files.exists(path) && time.isPresent()) {
-            res = Files.readAttributes(path, BasicFileAttributes.class)
-                .creationTime()
-                .toInstant()
-                .truncatedTo(ChronoUnit.MINUTES)
-                .equals(
-                    ZonedDateTime.parse(time.get()).toInstant().truncatedTo(ChronoUnit.MINUTES)
-                );
+        if (Files.exists(path) && hash.isPresent()) {
+            res = new XMLDocument(path).xpath("/program/@hash").stream().findFirst()
```

Comment:
@Yanich96 We use `OptCached` in order to avoid repeated unnecessary computations and increase the compilation speed. However the solution you provided might make the optimization even longer. Let's me explain. 

In order to read `/program/@hash` from the `path` you need to read the entire file. This operation is extremely long. I even bet that if you disable caching now, the compilation will be faster.

Reply:
@volodya-lombrozo 
Yes, but I have 3 points:
1. File won't be read from disk each time, thanks page cache https://habr.com/ru/companies/smart_soft/articles/228937/
2. In most languages reading from cache is faster, than compilation. Isn't it true for EO lang?
3. If reading from cache takes longer, then compilation.. so why the cache exists?

Reply:
@Yanich96, I conducted a simple benchmark to compare the time spent in `OptCache.apply` for the `returnsFromCacheIfXmlAlreadyInCache` test in the master branch (the old solution) and your solution. The results I obtained are as follows:

| Test Scenario                                  | Master Branch (ms)                    | Your Branch (ms)                      |
|------------------------------------------------|----------------------------------------|--------------------------------------|
| returnsFromCacheIfXmlAlreadyInCache Test       | 3654, 3680, 3693, 3700, 3862          | 4281, 4602, 4378, 4364, 4504    |


As you can see, the "page cache" doesn't provide much improvement. Please note that in the test, we are using an extremely small program example (only a few lines). Real programs will be much larger, and the performance gap will likely be more significant.

Below is the test code, where I modified the test to include time measurement.

Master branch:
```java
@Test
void returnsFromCacheIfXmlAlreadyInCache(@TempDir final Path tmp) throws IOException {
    final XML program = OptCachedTest.program(ZonedDateTime.now());
    OptCachedTest.save(tmp, program);
    final long start = System.currentTimeMillis();
    XML apply = null;
    for (int i = 0; i < 10_000; ++i) {
        apply = new OptCached(
            path -> {
                throw new IllegalStateException("This code shouldn't be executed");
            },
            tmp
        ).apply(program);
    }
    final long finish = System.currentTimeMillis();
    Logger.info(
        this,
        "Optimization took %d ms",
        finish - start
    );
    MatcherAssert.assertThat(
        "We expected that the program will be returned from the cache.",
        apply,
        Matchers.equalTo(program)
    );
}
```

Your branch:
```java
@Test
void returnsFromCacheIfXmlAlreadyInCache(@TempDir final Path tmp)
    throws IOException, ImpossibleModificationException, NoSuchAlgorithmException {
    final XML program = OptCachedTest.updatedProgram(
        OptCachedTest.program(ZonedDateTime.now())
    );
    OptCachedTest.save(tmp, program);
    final long start = System.currentTimeMillis();
    XML apply = null;
    for (int i = 0; i < 10_000; ++i) {
        apply = new OptCached(
            path -> {
                throw new IllegalStateException("This code shouldn't be executed");
            },
            tmp
        ).apply(program);
    }
    final long finish = System.currentTimeMillis();
    Logger.info(
        this,
        "Optimization took %d ms",
        finish - start
    );
    MatcherAssert.assertThat(
        "We expected that the program will be returned from the cache.",
        apply,
        Matchers.equalTo(program)
    );
}
```